junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ScriptExecutionCondition.java

```diff
@@ -111,66 +98,70 @@ private String createSource(String[] lines) {
-		Bindings bindings = new SimpleBindings();
-		bindings.put(Script.BIND_JUNIT_TAGS, context.getTags());
-		bindings.put(Script.BIND_JUNIT_UNIQUE_ID, context.getUniqueId());
-		bindings.put(Script.BIND_JUNIT_DISPLAY_NAME, context.getDisplayName());
-		bindings.put(Script.BIND_JUNIT_CONFIGURATION_PARAMETER, configurationParameterAccessor);
-		return bindings;
+	private Worker getScriptExecutionWorker(ExtensionContext context) {
+		ExtensionContext root = context.getRoot();
+		ExtensionContext.Store rootStore = root.getStore(NAMESPACE);
+		return rootStore.getOrComputeIfAbsent(WORKER, this::createWorker, Worker.class);
 	}
 
-	ConditionEvaluationResult evaluate(ScriptExecutionManager manager, Script script, Bindings bindings) {
-		if (script == null) {
-			return null;
+	// Create worker via reflection to hide the `javax.script` dependency.
+	private Worker createWorker(String name) {
+		logger.debug(() -> String.format("Creating instance of `%s`...", name));
+		Optional<Throwable> optionalThrowable = assumeScriptEngineIsLoadableByClassForName();
+		if (optionalThrowable.isPresent()) {
+			return new ThrowingWorker(optionalThrowable.get());
 		}
 		try {
-			Object result = manager.evaluate(script, bindings);
-			return computeConditionEvaluationResult(script, result);
+			return (Worker) Class.forName(name).getDeclaredConstructor().newInstance();
 		}
-		catch (ScriptException e) {
-			throw new ScriptEvaluationException("Script evaluation failed for: " + script.getAnnotationAsString(), e);
+		catch (ReflectiveOperationException cause) {
+			logger.error(cause, () -> String.format("Creating instance of `%s` failed", name));
+			return new ThrowingWorker(cause);
 		}
 	}
 
-	ConditionEvaluationResult computeConditionEvaluationResult(Script script, Object result) {
-		// Treat "null" result as an error.
-		if (result == null) {
-			throw new ScriptEvaluationException("Script returned `null`: " + script.getAnnotationAsString());
+	// This method may fail on JREs that don't provide the "javax.script" package at all
+	// or it fails on JREs launched with an active module system using insufficient module
+	// graphs, i.e. the application does not read "java.scripting" module.
+	Optional<Throwable> assumeScriptEngineIsLoadableByClassForName() {
+		try {
+			Class.forName("javax.script.ScriptEngine");
+			return Optional.empty();
 		}
-
-		// Trivial case: script returned a custom ConditionEvaluationResult instance.
-		if (result instanceof ConditionEvaluationResult) {
-			return (ConditionEvaluationResult) result;
+		catch (Throwable cause) {
+			logger.error(cause, () -> "Loading `javax.script.ScriptEngine` failed");
+			return Optional.of(cause);
 		}
+	}
 
-		String resultAsString = String.valueOf(result);
-		String reason = script.toReasonString(resultAsString);
+	/**
+	 * Evaluates scripts and returns a conditional evaluation result.
+	 */
+	interface Worker {
```

Comment:
From my perspective, the term `Worker` implies that we are doing something with forked processes or threads; whereas, as in your JavaDoc... this is something that "Evaluates scripts". 

So, I'd simply call it a `ScriptEvaluator` analogous to the existing `ConditionEvaluator`. 😉